std-bits/base/src/main/java/org/enso/base/parser/NumberWithSeparators.java

```diff
@@ -0,0 +1,348 @@
+package org.enso.base.parser;
+
+import java.util.Arrays;
+import org.enso.base.parser.FormatDetectingNumberParser.NumberParseDouble;
+import org.enso.base.parser.FormatDetectingNumberParser.NumberParseFailure;
+import org.enso.base.parser.FormatDetectingNumberParser.NumberParseLong;
+import org.enso.base.parser.FormatDetectingNumberParser.NumberParseResult;
+
+/**
+ * Number parsing with separators. Two special cases: - Encounter a single . or , with 3 trailing
+ * numbers. - Could be either DOT_COMMA or COMMA_DOT. Hence, DOT_UNKNOWN and COMMA_UNKNOWN.
+ */
+public enum NumberWithSeparators {
+  UNKNOWN(Constants.UNKNOWN, Constants.UNKNOWN),
+
+  // Special case where we have encountered a . with 3 trailing digits. ##0.123 ####.123
+  DOT_UNKNOWN(Constants.UNKNOWN, '.'),
+  // Special case where we have encountered a single . within 3 digits from start and without 3
+  // digits from end.
+  UNKNOWN_DOT(Constants.UNKNOWN, '.'),
```

Comment:
I really appreciate the examples like you given in the `DOT_UNKNOWN` case above. They make it much easier to understand the descriptions than trying to imagine them. Can we add one here as well?